### ISSUE TITLE
[Task]: Compute onPageLoad module actions

### ISSUE THREAD

**Issue by subrata71** [appsmithorg/appsmith#24636]
### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by subrata71** [comment]
### Outcome

TL;DR
--
We need to do a PoC

--------

In order to find on-page load module instances we need to take care of the following three cases. We will do a PoC on this and this activity will be tracked in this issue https://github.com/appsmithorg/appsmith/issues/25558

|Case|Example|Remarks|
-----|-----|----|
|Directly referenced module instance in the page DSL e.g. widgets|`Text1.text = {{ModuleInstance1.data}}`| `onPageLoadActions = [  ["ModuleInstance1"] ]`|
|User can explicitly set executeOnLoad although the module instance may not be directly referred from the page DSL|Go to settings tab of module instance and activate run on page load; ModuleInstance1.input = {{Api1.data}}, ModuleInstance1.executeOnLoad = true|`onPageLoadActions = [["Api1"], ["ModuleInstance1"]]`|
|Find transitive relationship with other actions, module instances and widgets| `Text1.text = {{Api1.data}}`, `Api1.body = {{ModuleInstance1.data}}`| ModuleInstance1 will also be considered as an on page load action; `onPageLoadActions = [["ModuleInstance1"], ["Api1"]]` |



![image](https://github.com/appsmithorg/appsmith/assets/3524599/6f469fab-268c-4c99-849d-e5522a07490e)
